slurmutils/core/editor.py

```diff
@@ -43,8 +43,13 @@ def dump(
         group: str | int | None = None,
     ) -> None:
         """Marshal a configuration model into a file."""
-        Path(file).write_text(self.dumps(obj) + "\n")
-        _set_file_permissions(file, mode=mode, user=user, group=group)
+        # Write to a temp file then atomically replace the target file.
```

Comment:
[nit]: This comment should be in the method docstring rather than in the method body. Some folks might not recognize the atomic operation, so we could just update the docstring to something like the following:

```python3
def dump(...) -> None:
    """Atomically marshal a configuration model into a file."""
```